includes/ingest.batch.inc

```diff
@@ -430,11 +432,20 @@ function islandora_batch_process_results($results, $timeout, $lock_timeout, &$co
         $context['message'] = t('Exception occured: Failed to ingest %pid.', array('%pid' => $ingest_object->id));
       }
     }
+    else {
+      $context['message'] = t('%pid not ready for ingest.', array('%pid' => $ingest_object->id));
+    }
 
     // Update the info in the database.
     module_load_include('inc', 'islandora_batch', 'includes/db');
     islandora_batch_merge_queued_object_data($object['id'], $object['data']);
     islandora_batch_merge_object_state($object['id'], $object['state']);
+    islandora_batch_merge_object_message(
+      $object['id'],
+      isset($process_results['message']) && $process_results['message'] ?
+        $process_results['message'] :
+        strip_tags($context['message'])
```

Comment:
Not sure what to say here... Usually don't sanitize on the way into the DB, it's normally left for the way out. Using the DB-API should be proof against SQL injection... Dunno.